backend/state.py

```diff
@@ -463,6 +463,13 @@ def step(self, actions, clock):
             AssertionError: If the action is invalid with the given arguments in
             the given state.
         """
+
+        for special_effect in self.special_effects:
```

Comment:
PR note to self - this was done to handle an off by one error with rendering. gameplay unaffected